ssas/service/logging.go

```diff
@@ -74,8 +79,23 @@ func NewCtxLogger(next http.Handler) http.Handler {
 		if rd, ok := r.Context().Value("rd").(ssas.AuthRegData); ok {
 			logFields["okta_id"] = rd.OktaID
 		}
+		logFields["transaction_id"] = r.Context().Value(CtxTransactionKey).(string)
 		newLogEntry := &log.APILoggerEntry{Logger: log.Logger.WithFields(logFields)}
 		r = r.WithContext(context.WithValue(r.Context(), log.CtxLoggerKey, newLogEntry))
 		next.ServeHTTP(w, r)
 	})
 }
+
+// type to create context.Context key
+type CtxTransactionKeyType string
+
+// context.Context key to get the transaction ID from the request context
+const CtxTransactionKey CtxTransactionKeyType = "ctxTransaction"
+
+// Adds a transaction ID to the request context
+func NewTransactionID(next http.Handler) http.Handler {
```

Comment:
For this ticket, instead of generating a new transaction ID, we need to get the transactionID from the request headers from the BCDA-App request. We could leave this function in here and add transaction IDs to log statements for all of the endpoints in a future ticket. This allows us to correlate logs between the two applications.